src/types.ts

```diff
@@ -0,0 +1,45 @@
+import { HierarchyRectangularNode } from "d3";
+
+/**
+ * Implied properties for each JSON object in `hierarchy.json`
+ */
+export type Datum = {
+  name: string;
+  value?: number;
+  label?: string;
+  children?: Datum[];
+};
+
+/**
+ * Properties extended by {@link HierarchyRectangularNode}
+ */
+export type Rectangle = {
+  /**
+   * The left edge of the rectangle.
+   */
+  x0: number;
+
+  /**
+   * The top edge of the rectangle
+   */
+  y0: number;
+
+  /**
+   * The right edge of the rectangle.
+   */
+  x1: number;
+
+  /**
+   * The bottom edge of the rectangle.
+   */
+  y1: number;
+};
```

Comment:
TODO after migration: determine what each of these mean in the sunburst (we do not use them as commented)